lib/resty/core/shdict.lua

```diff
@@ -387,6 +406,34 @@ local function shdict_flush_all(zone)
 end
 
 
+local function shdict_stats(zone, slots)
+    zone = check_zone(zone)
+
+    C.ngx_http_lua_ffi_shdict_get_stats_slot_count(zone, slot_count_buf)
+    local slot_count = tonumber(slot_count_buf[0])
+    local raw_stats_buf = get_string_buf(slot_count * slab_stat_t_size)
+    local stats_buf = ffi_cast(slab_stat_t_ptr_type, raw_stats_buf)
+    C.ngx_http_lua_ffi_shdict_get_stats_slots_and_min_size(zone, slot_count_buf[0], stats_buf, slot_min_size_buf)
+    local slot_size = tonumber(slot_min_size_buf[0])
+    local total_used = 0
+    for i = 1, slot_count do
+        local stat = stats_buf[i - 1]
+        if slots then
+            slots[i] = {
+                size = slot_size,
+                total = tonumber(stat.total),
+                used = tonumber(stat.used),
+                reqs = tonumber(stat.reqs),
+                fails = tonumber(stat.fails),
+            }
+        end
+        total_used = total_used + slot_size * tonumber(stat.used)
+        slot_size = slot_size * 2
+    end
+    return total_used
```

Comment:
Let's just return the total used and total size values directly? This way we do not need the dynamically-sized cdata or Lua tables.